### PR TITLE
Add docs about shipping and billing addresses validation

### PR DESCRIPTION
**Pull request by korycins**
Docs for PRs https://github.com/saleor/saleor/pull/10084 & https://github.com/saleor/saleor/pull/10102

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-7545 Docs for address validations](https://app.clickup.com/t/2549495/SALEOR-7545)

**Comment by peelar** [comment]
@korycins I see this one has been stuck here for quite some time. I would love to help with reviewing and pushing that, but it appears there are many conflicts here. Can we do something about it?

**Comment by korycins** [comment]
@peelar  Conflicts solved. 🤗

File: docs/developer/api-reference/enums/webhook-event-type-async-enum.mdx [hunk 1]
```diff
@@ -433,7 +433,7 @@ A shipping zone is deleted.
 
 #### [`STAFF_CREATED`](#)
 
-A new staff user is created.
+A staff user is deleted
```
Comment (peelar) [inline docs/developer/api-reference/enums/webhook-event-type-async-enum.mdx:436]: > A staff user is deleted

The event is called `STAFF_CREATED` but the content is "A staff user is deleted". Is that right?

File: docs/developer/api-reference/enums/webhook-event-type-enum.mdx [hunk 2]
```diff
@@ -445,7 +445,7 @@ A shipping zone is deleted.
 
 #### [`STAFF_CREATED`](#)
 
-A new staff user is created.
+A staff user is deleted
```
Comment (peelar) [inline docs/developer/api-reference/enums/webhook-event-type-enum.mdx:448]: Same as above 🤔.

File: docs/developer/checkout.mdx [hunk 3]
```diff
@@ -39,6 +39,176 @@ Depending on the chosen channel, the user will have access to different objects.
 
 If the checkout permission is assigned to the user, only this user and staff users with permission `MANAGE_CHECKOUTS` can query its data. Checkouts without assigned users can be queried modified using the token without additional permissions.
 
+## Address Validation
+The checkout's mutations that accepts an address as an input have field that can turn off address validation. This allows to assign partial or not fully valid address to the checkout. Providing country code is mandatory for all addresses regardless of the rules provided in this input.
```
Comment (peelar) [inline docs/developer/checkout.mdx:43]:
```suggestion
The checkout's mutations that accept an address as an input have a field that can turn off the address validation. This allows assigning a partial or not fully valid address to the checkout. Providing country code is mandatory for all addresses regardless of the rules provided in this input.
```

File: docs/developer/checkout.mdx [hunk 4]
```diff
@@ -39,6 +39,176 @@ Depending on the chosen channel, the user will have access to different objects.
...
+
+The address [validation](developer/api-reference/inputs/checkout-address-validation-rules.mdx) input has two boolean fields: 
+- `checkRequiredFields` - Determines if an error should be raised when the provided address doesn't have all the required fields. Default set to true.
+- `checkFieldsFormat` - Determines if an error should be raised when the provided address doesn't match the expected format. Default set to true.
+- `enableFieldsNormalization` - Determines if Saleor should apply normalization on address fields. Example: converting city field to uppercase letters. Default set to true.
```
Comment (peelar) [inline docs/developer/checkout.mdx:48]:
```suggestion
- `checkRequiredFields` - Determines if an error should be raised when the provided address doesn't have all the required fields. Set to `true` by default.
- `checkFieldsFormat` - Determines if an error should be raised when the provided address doesn't match the expected format. Set to `true` by default.
- `enableFieldsNormalization` - Determines if Saleor should apply normalization on address fields. Example: converting city field to uppercase letters. Set to `true` by default.
```

File: docs/developer/checkout.mdx [hunk 5]
```diff
@@ -39,6 +39,176 @@ Depending on the chosen channel, the user will have access to different objects.
...
+
+#### checkoutCreate
+The [`checkoutCreate`](developer/api-reference/mutations/checkout-create.mdx) mutation has optional input for providing shipping and billing addresses. By disabling an address validation there is a possibility to provide only part of an address during checkout creation.
```
Comment (peelar) [inline docs/developer/checkout.mdx:51]:
```suggestion
The mutation has an optional input for providing shipping and billing addresses. If you want to provide only a part of the address, you can disable the address validation.
```

File: docs/developer/checkout.mdx [hunk 6]
```diff
@@ -39,6 +39,176 @@ Depending on the chosen channel, the user will have access to different objects.
...
+      },
+      "errors": []
+    }
+  },
+}
+```
+
+
+#### CheckoutShippingAddressUpdate
+The [`checkoutShippingAddressUpdate`](developer/api-reference/mutations/checkout-shipping-address-update.mdx) mutation has optional field for controling a shipping address validation: [validationRules](developer/api-reference/inputs/checkout-address-validation-rules.mdx).
```
Comment (peelar) [inline docs/developer/checkout.mdx:96]:
```suggestion
The [`checkoutShippingAddressUpdate`](developer/api-reference/mutations/checkout-shipping-address-update.mdx) mutation has an optional field for controlling the shipping address validation: [validationRules](developer/api-reference/inputs/checkout-address-validation-rules.mdx).
```

File: docs/developer/checkout.mdx [hunk 7]
```diff
@@ -39,6 +39,176 @@ Depending on the chosen channel, the user will have access to different objects.
...
+          }
+        }
+      }
+    }
+  },
+}
+```
+
+#### CheckoutBillingAddressUpdate
+The [`checkoutBillingAddressUpdate`](developer/api-reference/mutations/checkout-billing-address-update.mdx) mutation has optional field for controling a billing address validation: [validationRules](developer/api-reference/inputs/checkout-address-validation-rules.mdx).
```
Comment (peelar) [inline docs/developer/checkout.mdx:150]:
```suggestion
The [`checkoutBillingAddressUpdate`](developer/api-reference/mutations/checkout-billing-address-update.mdx) mutation has an optional field for controlling the billing address validation: [validationRules](developer/api-reference/inputs/checkout-address-validation-rules.mdx).
```

File: docs/developer/api-reference/enums/webhook-event-type-async-enum.mdx (hunk 1 above)
Comment (korycins) [reply docs/developer/api-reference/enums/webhook-event-type-async-enum.mdx:436]: Hhm... Thanks for catching this. I will check this.

File: docs/developer/checkout.mdx (hunk 5 above)
Comment (korycins) [reply docs/developer/checkout.mdx:51]: @peelar  Wouldn't be good to leave api-reference here? Something like:
`The [mutation]((developer/api-reference/mutations/checkout-create.mdx)) has an optional...`

Comment (peelar) [reply docs/developer/checkout.mdx:51]: Ah shit! Sorry, I removed it by accident. It made Grammarly go crazy 😵. Fixing it

Comment (peelar) [reply docs/developer/checkout.mdx:51]: Ah shit! Sorry, I removed it temporarily because it made Grammarly go crazy but I forgot to switch it back. Fixed!

**Comment by korycins** [comment]
@peelar Thanks for the help! All requested changes have been applied

**Comment by peelar** [comment]
@korycins you are on fire today 🔥 🔥 🔥!

**Review by peelar — APPROVED** [review]

**Pull request merged**